pkg/controller/nodeipam/ipam/cloud_cidr_allocator_test.go

```diff
@@ -83,3 +83,43 @@ func TestNodeUpdateRetryTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateCIDRs(t *testing.T) {
+	for _, tc := range []struct {
+		desc    string
+		cidrs   []string
+		wantErr bool
+	}{
+		{
+			desc:    "Valid IPV4 cidrs",
+			cidrs:   []string{"10.10.10.1/24", "172.168.1.2/32"},
+			wantErr: false,
+		},
+		{
+			desc:    "Invalid IPV4 cidrs",
+			cidrs:   []string{"10.10.10.1/24", "172.168..1.2/32"},
+			wantErr: true,
+		},
+		{
+			desc:    "Valid IPV6 cidrs",
+			cidrs:   []string{"2001:db8::/64", "::1234:5678/128"},
+			wantErr: false,
+		},
+		{
+			desc:    "Invalid IPV6 cidrs",
+			cidrs:   []string{"2001:db8::", "::1234:5678/130"},
+			wantErr: true,
+		},
+		{
+			desc:    "Valid cidrs",
+			cidrs:   []string{"10.10.10.1/24", "2001:db8::/64"},
+			wantErr: false,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
```

Comment:
If we can, we should check the node.Spec being set correctly.

Reply:
I wanted to add unit tests for updateCIDRAllocation but I couldn't find a straightforward way to mock ca.cloud.AliasRangesByProviderID() as it not implemented as an interface that can be mocked or stubbed.